system/lib/libc/raise.c

```diff
@@ -20,34 +20,39 @@ bool __sig_is_blocked(int sig);
 // Default handler actions ~auto-generated from https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/signal.h.html.
 // Note that stop and continue actions are not supported and treated as ignored.
 
-typedef enum {
-  ACTION_IGNORE = 0,
-  ACTION_TERMINATE,
-  ACTION_ABORT,
-} default_action_t;
+void action_abort(int sig) {
+  abort();
+}
+
+void action_terminate(int sig) {
+  // Prepare to forcibly shut down runtime even if it has async work in flight.
+  _emscripten_runtime_keepalive_clear();
```

Comment:
What if we didn't do this part, at least for the first version of this PR?   Would it negatively effect your use case to let the async work complete?

Reply:
then would could avoid adding this new API or modifying `src/library.js` as part of this PR, keeping it very focused?

If it turns out we real do want to ignore outstanding work, that could be a followup?

Reply:
Hm but this _the_ point of this PR TBH. The main issue I ran into with previous implementation and tried to fix is that terminate action didn't do anything when there was async work in progress.

Reply:
To describe specific scenario more precisely, the problem is that in PTY integration onSignal is a separate event from asynchronous reading.

If an application is waiting in a readline, and user presses Ctrl+C, my onSignal handler will `raise(SIGINT)` in its Emscripten integration, but the app won't stop as expected because it still has that async task waiting for a complete line in background.

I could in theory somehow communicate between onSignal and readline, but that would only fix one particular async API, whereas a fix from this PR covers them all together.

Reply:
Ok, fair enough.